### PR TITLE
[tests-only][full-ci]Add api tests for changing role of a space

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
This PR adds api tests for changing the role of a member of the space

## Related Issue

- Part of : https://github.com/owncloud/ocis/issues/4937

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by SwikritiT** [comment]
@ScharfViktor while changing the role of the members of shared space `POST` request is sent just like while sharing the spaces and the payload is also the same. I'm confused should it be `PUT` or `POST` is the correct one? I noticed foe other shares like normal file sharing `update` related operations is done through `PUT`

File: tests/acceptance/features/apiSpacesShares/shareSpaces.feature [hunk 1]
```diff
@@ -136,3 +136,32 @@ Feature: Share spaces
...
+    When user "Alice" changes the role of user "Brian" to "<new_role>" for space "share space"
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "200"
+    Then the user "Alice" should have a space called "share space" granted to "Brian" with role "<new_role>"
+    Examples:
+      | role   | new_role |
+      | editor | manager  |
+      | editor | viewer   |
+      | viewer | manager  |
+      | viewer | editor   |
```
[the remark this answers is not in the document]
Comment (SwikritiT) [reply tests/acceptance/features/apiSpacesShares/shareSpaces.feature:152]: I'll add that example !

[the remark this answers is not in the document]
Comment (SwikritiT) [reply tests/acceptance/features/apiSpacesShares/shareSpaces.feature:152]: @ScharfViktor I also mentioned you here https://github.com/owncloud/ocis/pull/5088#issuecomment-1321815533 I didn't know if you've noticed it or not

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -1898,6 +1897,21 @@ public function sendShareSpaceRequest(
 		);
 	}
 
+	/**
+	 * @When /^user "([^"]*)" changes the role of user "([^"]*)" to "([^"]*)" for space "([^"]*)"$/
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1901]: there is already a step for the func. `sendShareSpaceRequest` imo, u can reuse that.

**Review by kiranparajuli589 — COMMENTED** [review]
rest lgtm

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 3]
```diff
@@ -1753,7 +1753,6 @@ public function userShouldNotBeAbleToDownloadFileInsideSpace(
 		string $spaceName
 	):void {
 		$space = $this->getSpaceByName($user, $spaceName);
-		$fullUrl = $space["root"]["webDavUrl"] . '/' . ltrim($fileName, "/");
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1756]: `$space` is not used as well, is it?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 4]
```diff
@@ -2053,6 +2052,7 @@ public function userHasCreatedPublicLinkToEntityInsideOfSpaceRequest(
 
 	/**
 	 * @Given /^user "([^"]*)" has shared a space "([^"]*)" to user "([^"]*)" with role "([^"]*)"$/
+	 * @When /^user "([^"]*)" updates the space "([^"]*)" for user "([^"]*)" changing the role to "([^"]*)"$/
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2055]: @kiranparajuli589 did you mean this in your previous review https://github.com/owncloud/ocis/pull/5088#discussion_r1028841057?

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/SpacesContext.php:2055]: yes :+1:

File: tests/acceptance/features/apiSpacesShares/shareSpaces.feature [hunk 5]
```diff
@@ -136,3 +136,34 @@ Feature: Share spaces
...
+      | manager | viewer   |
+
+
+  Scenario Outline: user without manager role cannot change the role of space members
+    Given user "Alice" has shared a space "share space" to user "Brian" with role "<role>"
+    And user "Alice" has shared a space "share space" to user "Bob" with role "viewer"
+    When user "Brian" updates the space "share space" for user "Bob" changing the role to "<new_role>"
+    Then the HTTP status code should be "404"
+    And the OCS status code should be "404"
+    Then the user "Alice" should have a space called "share space" granted to "Bob" with role "viewer"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSpacesShares/shareSpaces.feature:163]:
```suggestion
    And the user "Alice" should have a space called "share space" granted to "Bob" with role "viewer"
```

File: tests/acceptance/features/apiSpacesShares/shareSpaces.feature [hunk 6]
```diff
@@ -136,3 +136,34 @@ Feature: Share spaces
...
       | viewer | editor   |
       | viewer | viewer   |
+
+
+  Scenario Outline: space manager can change the role of space members
+    Given user "Alice" has shared a space "share space" to user "Brian" with role "<role>"
+    When user "Alice" updates the space "share space" for user "Brian" changing the role to "<new_role>"
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "200"
+    Then the user "Alice" should have a space called "share space" granted to "Brian" with role "<new_role>"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSpacesShares/shareSpaces.feature:146]:
```suggestion
    And the user "Alice" should have a space called "share space" granted to "Brian" with role "<new_role>"
```

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

File: tests/acceptance/features/apiSpacesShares/shareSpaces.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply tests/acceptance/features/apiSpacesShares/shareSpaces.feature]: I propose to leave it as it is. not critical for my opinion

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**